File: frontend/src/app/models/book-item.ts
import { Book } from "./book";
import { BookItemFormat } from "../shared/book-item-format";
import { BookItemStatus } from "../shared/book-item-status";
import { Rack } from "./rack";

export class BookItem {
    id: number;
    barcode: string;
    isReferenceOnly: boolean;
    borrowed: Date;
    dueDate: Date;
    price: BigInt;
    format: BookItemFormat;
    status: BookItemStatus;
    dateOfPurchase: Date;
    publicationDate: Date;
    book: Book;
    rack: Rack;
}

File: backend/src/main/java/com/example/libraryapp/domain/bookitem/model/BookItem.java
package com.example.libraryapp.domain.bookitem.model;

import com.example.libraryapp.domain.book.model.BookId;
import com.example.libraryapp.domain.bookitemloan.model.LoanCreationDate;
import com.example.libraryapp.domain.bookitemloan.model.LoanDueDate;
import com.example.libraryapp.domain.rack.model.RackId;
import lombok.*;

@Getter
@Setter
@Builder
@AllArgsConstructor
@RequiredArgsConstructor
public class BookItem {
    private BookItemId id;
    private BookItemBarcode barcode;
    private IsReferenceOnly isReferenceOnly;
    private LoanCreationDate borrowedDate;
    private LoanDueDate dueDate;
    private Price price;
    private BookItemFormat format;
    private BookItemStatus status;
    private PurchaseDate dateOfPurchase;
    private PublicationDate publicationDate;
    private BookId bookId;
    private RackId rackId;
}
